docs/concepts/v2/concepts/rooms/room-codes/room-code-auth.mdx

```diff
@@ -3,24 +3,184 @@ title: Room Code Authentication
-1. Retrieve Room Code - You can get the Room Code in two ways:
+When building a custom UI and using room codes to generate room links, you can also use room codes to power authentication to join room without setting up an auth token server at your end.  
 
-    - Retrieve the Room Code from the [Room Link](/concepts/v2/concepts/rooms/room-links) that you generate.
-    - Use [Get Room Code API](/server-side/v2/api-reference/room-codes/get-room-code-api) to retrieve the Room Code
+### Implementation
 
-2. You need to then pass the Room Code in the join() method in the SDK. Here, the SDK implicitly generates Authentication Token from the Room Code.
+1. **Create Room Codes** - Room codes can be made available in two ways:
 
-### Room Code Authentication for Prebuilt
+<Tabs id="implementation" items={['via Dashboard','Create Room Code API']} />
 
-If you're using 100ms Prebuilt solution, you need not handle authentication as this is implicitly handled within Prebuilt.
\ No newline at end of file
+    <Tab id='implementation-0'>
+
+    For Rooms created on [100ms dashboard](https://dashboard.100ms.live/dashboard), a room code gets generated for every role in the room by default.
+
+    ![RoomCodeFromJoinRoomModal](/docs/v2/JoinRoomModal.png)
+
+    </Tab>
+
+
+    <Tab id="implementation-1">
+
+    If you're using [Create Room API](/server-side/v2/api-reference/Rooms/create-via-api) to create Rooms, you need to explicitly create room codes for every role in the room. Refer [Create Room Codes API](/server-side/v2/api-reference/room-codes/create-room-code-api). This API allows you to create Room Code for
+             - every role in the room at once  using [Create Room Code API](/server-side/v2/api-reference/room-codes/create-room-code-api#1-create-a-room-code-api) 
+             - a specific role in the room using [Create Role Room Code API](server-side/v2/api-reference/room-codes/create-room-code-api#2-create-role-room-code-api)
+
+    </Tab>
+
+2. **Retrieve role specific room code** - you can get the room code in two ways:
+
+    - **No-Code** 
+        - Select a room from Room page on [100ms Dashboard](https://dashboard.100ms.live/rooms) and click on "Join Room"
+        - Copy room code for the role you want to authenticate
+    ![RoomCodeFromJoinRoomModal](/docs/v2/JoinRoomModal.png)
+       
+        
+
+    - Programmatically,
+        - From the [Room Link](/concepts/v2/concepts/rooms/room-links) that you generate. For instance, you can retrieve room code from a custom room link ` https://<your-template-subdomain>.app.100ms.live/meeting/<room-code> `
+        - Alternatively, you can use [Get Room Code API](/server-side/v2/api-reference/room-codes/get-room-code-api) to retrieve the room code for the role joining the room.
+
+3. Once you have a role specific room code, call the SDK function `getAuthTokenByRoomCode` with the retrieved room code. This function retrieves the auth token that is required to join the room.
+
+
+##### Sample Code 
```

Comment:
i think it'll be useful to have a heading here to say "room codes in custom UI" - similar to what you have in the callout on this page above